src/charm.py

```diff
@@ -173,8 +175,21 @@ def apply_kube_ovn(self, service_cidr, registry):
         cni_server_container = self.get_container_resource(
             kube_ovn_cni, container_name="cni-server"
         )
-        self.replace_container_args(
-            cni_server_container, args={"--service-cluster-ip-range": service_cidr}
+
+        cni_args_to_replace = {"--service-cluster-ip-range": service_cidr}
+        if enable_global_mirror and not mirror_iface:
+            log.error("If enable-global-mirror is true, mirror-iface must be set")
+            self.unit.status = BlockedStatus(
+                "If enable-global-mirror is true, mirror-iface must be set"
+            )
```

Comment:
doesn't this  need a `return` or something to prevent it from continuing on after this blocking condition?
or is the cni okay to have `--enable-mirror=false --mirror-iface=''`?  

I believe you're saying that it's okay where in your summary you state this:

> If enable-global-mirror is true, then the mirror-iface config option cannot be empty. If it is, the charm will block with a message alerting the user to the misconfiguration, and the cni-server will be configured without the mirror enabled.

Reply:
thats a good point. By default, if you dont give the --mirror-iface arg to the binary, it will just use mirror0, but if you provide an empty string it might try and use that. I could return there, but it might be better to just block the charm, but still go on and just pass mirror0 to be explicit

Reply:
i mean, if you return -- none of the other business get's to setup.  is that okay?  I haven no idea.

if you just let it carry on -- maybe don't set "--mirror-iface" from the config.

Reply:
alright, so here is what happens:
If the user sets enablle-global-mirror to true but sets mirror-iface to "", then we block and alert the user that this is not valid and that a iface must be set by specifying a non empty value in config (or relying on the charm config default of mirror0 if not specified at all)

The mirror interface can be set independently of enable-global-mirror (such as when doing non-global-mirroring), but I have added an if statement to only set this cni option if the interface is non-empty. This way we no longer can get something like `--mirror-iface=''`. When this option is not specified at all, kube-ovn defaults to using mirror0 as the interface as seen [here](https://github.com/kubeovn/kube-ovn/blob/9ff3b9c05983b7b88b8fb54a526ee197bd0326cf/pkg/daemon/config.go#L69)